app/views/admin/collections/_form.html.erb

```diff
@@ -0,0 +1,24 @@
+<div class="deal-form">
+  <%= form_for [:admin, collection] do |form| %>
+    <div class="form-container">
+
+      <div class="field">
+        <%= form.label :name, class: 'col-form-label' %>
+        <%= form.text_field :name, class: 'form-control', placeholder: 'Enter name' %>
+      </div>
+
+      <div class="field">
+        <%= form.label 'Deals', class: 'col-form-label' %>
+        <%= form.collection_select :deal_ids, Deal.where(collection_id: nil, published_at: nil).or(Deal.where(collection_id: collection.id)), :id, :title, {}, multiple: true, class: 'form-control', data: { type: 'select2-enabled' } %>
```

Comment:
use helper for Deal.where(collection_id: nil, published_at: nil).or(Deal.where(collection_id: collection.id), please do this in your current PR so that we can merge this one